backend/src/Designer/Infrastructure/GitRepository/AltinnAppGitRepository.cs

```diff
@@ -703,42 +703,79 @@ public async Task<string> GetAppFrontendCshtml(CancellationToken cancellationTok
+            cancellationToken.ThrowIfCancellationRequested();
+
+            string optionsFilePath = Path.Combine(OptionsFolderPath, $"{optionListId}.json");
             if (!FileExistsByRelativePath(optionsFilePath))
             {
-                throw new NotFoundException("Options file not found.");
+                throw new NotFoundException($"Options file {optionListId}.json was not found.");
             }
             string fileContent = await ReadTextByRelativePathAsync(optionsFilePath, cancellationToken);
 
             return fileContent;
         }
 
         /// <summary>
-        /// Gets a list of file names from the Options folder representing the available options lists.
-        /// <returns>A list of option list names.</returns>
+        /// Creates a new option list with the provided id.
+        /// If the option list already exists, it will be overwritten.
         /// </summary>
-        public string[] GetOptionListIds()
+        /// <param name="optionListId">The name of the option list to create.</param>
+        /// <param name="payload">The contents of the new option list as a string</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that observes if operation is cancelled.</param>
+        /// <returns>The new options list as a string.</returns>
+        public async Task<string> CreateOrOverwriteOptions(string optionListId, string payload, CancellationToken cancellationToken = default)
         {
-            string optionsFolder = Path.Combine(OptionsFolderPath);
-            if (!DirectoryExistsByRelativePath(optionsFolder))
-            {
-                throw new NotFoundException("Options folder not found.");
-            }
-            string[] fileNames = GetFilesByRelativeDirectory(optionsFolder);
-            List<string> optionListIds = new();
-            foreach (string fileName in fileNames.Select(Path.GetFileNameWithoutExtension))
+            cancellationToken.ThrowIfCancellationRequested();
+
+            string optionsFilePath = Path.Combine(OptionsFolderPath, $"{optionListId}.json");
+            await WriteTextByRelativePathAsync(optionsFilePath, payload, true, cancellationToken);
+            string fileContent = await ReadTextByRelativePathAsync(optionsFilePath, cancellationToken);
+
+            return fileContent;
+        }
+
+        /// <summary>
+        /// Deletes the option list with the provided id.
+        /// </summary>
+        /// <param name="optionListId">The name of the option list to create.</param>
+        public void DeleteOptions(string optionListId)
+        {
+            string optionsFilePath = Path.Combine(OptionsFolderPath, $"{optionListId}.json");
+            if (!FileExistsByRelativePath(optionsFilePath))
             {
-                optionListIds.Add(fileName);
+                throw new NotFoundException($"Options file {optionListId}.json was not found.");
```

Comment:
In this case I have removed the exception from the controller, which now returns 200 whether the file is found or not. I think maybe we'll keep this error handling here, just in case some future endpoint/services wants to use the delete method in the repository